### PR TITLE
pkp/pkp-lib#4789 Due date fields added for declined review request resending

### PR DESCRIPTION
**Pull request by touhidurabir**
for pkp/pkp-lib#4789 . see more details at https://github.com/pkp/pkp-lib/issues/4789#issuecomment-2014034378

### CODE REVIEW

File: controllers/grid/users/reviewer/form/ResendRequestReviewerForm.php [hunk 1]
```diff
@@ -45,21 +44,54 @@ public function __construct(ReviewAssignment $reviewAssignment, ReviewRound $rev
             $submission,
             'controllers/grid/users/reviewer/form/resendRequestReviewerForm.tpl'
         );
+
+        // Validation checks for this form
+        $this->addCheck(new \PKP\form\validation\FormValidator($this, 'responseDueDate', 'required', 'editor.review.errorAddingReviewer'));
+        $this->addCheck(new \PKP\form\validation\FormValidator($this, 'reviewDueDate', 'required', 'editor.review.errorAddingReviewer'));
```
Comment (Vitaliy-1) [inline controllers/grid/users/reviewer/form/ResendRequestReviewerForm.php:50]: Should we also add `FormValidatorPost` to the parent class? (not sure about that)

**Review by Vitaliy-1 — COMMENTED** [review]
Looks good! Just a comment regarding coding style

**Comment by touhidurabir** [comment]
@Vitaliy-1 sorry for late as I was waiting for the pkp/pkp-lib#5885 to get merged first . Please take a look again . BTW just wondering should we add some E2E tests for the review resend functionality ?

File: controllers/grid/users/reviewer/form/traits/HasReviewDueDate.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+namespace PKP\controllers\grid\users\reviewer\form\traits;
+
+use Carbon\Carbon;
+use PKP\context\Context;
+
+trait HasReviewDueDate
+{
+    public const REVIEW_SUBMIT_DEFAULT_DUE_WEEKS = 4;
+    public const RESPONSE_RESPONSE_DEFAULT_DUE_WEEKS = 4;
```
Comment (Vitaliy-1) [inline controllers/grid/users/reviewer/form/traits/HasReviewDueDate.php:23]: Probably should be named `REVIEW_RESPONSE_DEFAULT_DUE_WEEKS`. Also I see that it was 3 before, is it intentional?

**Review by Vitaliy-1 — COMMENTED** [review]
Thanks, I've added a comment.

File: controllers/grid/users/reviewer/form/traits/HasReviewDueDate.php (hunk 2 above)
Comment (touhidurabir) [reply controllers/grid/users/reviewer/form/traits/HasReviewDueDate.php]: Right , fixed it .

**Pull request merged**